pkg/planner/core/preprocess.go

```diff
@@ -762,6 +762,19 @@ func (p *preprocessor) checkAutoIncrement(stmt *ast.CreateTableStmt) {
 		case mysql.TypeTiny, mysql.TypeShort, mysql.TypeLong,
 			mysql.TypeFloat, mysql.TypeDouble, mysql.TypeLonglong, mysql.TypeInt24:
 		default:
+			if col.Tp.GetType() == mysql.TypeDate {
+				var defaultValueOpt *ast.ColumnOption
+				for _, op := range col.Options {
+					if op.Tp == ast.ColumnOptionDefaultValue {
+						defaultValueOpt = op
+						break
+					}
+				}
+				if defaultValueOpt != nil && defaultValueOpt.Expr != nil {
```

Comment:
I think we put this check to `checkAutoIncrementOp` is better. Because this function checks some `Invalid default value for ...` errors.

Reply:
I have made the changes as requested, could you please review it again?